client/components/Users/UserInfo.jsx

```diff
@@ -35,37 +38,75 @@ export default class UserInfo extends Component {
     return user.get('blocked') ? 'Yes' : 'No';
   }
 
-  renderUsername(username) {
-    if (!username) {
+  findprop(obj, path) {
+    var args = path.split('.'), i, l;
+
+    for (i=0, l=args.length; i<l; i++) {
+      if (!obj.hasOwnProperty(args[i]))
+        return;
+      obj = obj[args[i]];
+    }
+
+    return obj;
+  }
+
+  getValue(user, field) {
+    if (user.size === 0) {
       return null;
     }
 
-    return (
-      <UserInfoField title="Username">{username}</UserInfoField>
-    );
+    let value = this.findprop(user, field.property);
+    if (!value) return null;
+
+    if (field.type && field.type === 'elapsedTime') {
+      value = moment(value).fromNow();
+    }
+
+    return value;
   }
 
   render() {
     const { user, error, loading, memberships } = this.props;
-    const currentMemberships = this.getMemberships(memberships);
-    const identity = this.getIdentities(user);
-    const blocked = this.getBlocked(user);
+    const excludeCustomFields = ( this.props.userInfoFields && this.props.userInfoFields.excludeFields ) || [];
```

Comment:
This is starting to become a lot of logic to execute for every render. Should be moved to a higher level.

Reply:
Fixed in next version